plugins/techdocs/src/reader/components/TechDocsSearch.tsx

```diff
@@ -87,6 +88,17 @@ const TechDocsSearchBar = ({
 
   useDebounce(() => setTerm(value), debounceTime, [value]);
 
+  // Update the filter context when the entityId changes, e.g. when the search
+  // bar continues to be rendered, navigating between different TechDocs sites.
+  useEffect(() => {
+    setFilters(prevFilters => {
+      return {
+        ...prevFilters,
+        ...entityId,
+      };
+    });
+  }, [entityId, setFilters]);
```

Comment:
Is it safe to compare the entity id by reference? As opposed to e.g. its stringifyEntityRef version

Reply:
Yeah, I wonder if it actually makes the effect more readable to destruct the bits we use from it and merge those into the filter manually anyway.

Reply:
I like it!